packages/taquito-michelson-encoder/test/sample21_complex-bigmap.spec.ts

```diff
@@ -0,0 +1,158 @@
+import { Parser } from '@taquito/michel-codec'
+import { Schema } from '../src/schema/storage';
+
+describe('Complex big_map storage', () => {
+  it('Should extract correct bigmap schema', () => {
+    const example_storage_in_michelson = `(pair
+    (pair
+    (pair (address %administrator)
+    (big_map %balances address
+    (pair (map %approvals address nat) (nat %balance))))
+    (pair (nat %counter) (pair (nat %default_expiry) (nat %max_expiry))))
+    (pair
+    (pair (big_map %metadata string bytes)
+    (pair (bool %paused)
+    (big_map %permit_expiries (pair address bytes) (option nat))))
+    (pair (big_map %permits (pair address bytes) timestamp)
+    (pair (nat %totalSupply) (big_map %user_expiries address (option nat))))))`
+    
+    const p = new Parser();
+    const parsed_storage: any = p.parseMichelineExpression(example_storage_in_michelson);
```

Comment:
Can we remove the dependence on the michel-codec package by creating the Schema object directly from the JSON Michelson format?

Reply:
Done.